lib/fs_utils/generate.js

```diff
@@ -122,12 +122,21 @@ const runOptimizer = (optimizer, params) => {
   const optimizerArgs = optimizer.optimize.length === 2 || optimizer.optimize.length === 1 ?
     [params] : [data, path];
 
-  return helpers.promisifyPlugin(1, optimizer.optimize)
-    .apply(optimizer, optimizerArgs)
+  const _optimize = () => {
+    return helpers.promisifyPlugin(1, optimizer.optimize)
+      .apply(optimizer, optimizerArgs);
+  };
+
+  const scheduleOptimize = () => {
+    return global.wrkr.schedule('optimize', {optimizer: optimizer.constructor.brunchPluginName, params: {data,path}})
+  };
+
+  return (global.wrkr ? scheduleOptimize : _optimize)()
```

Comment:
Choose between original implementation and scheduling a worker job, depending on whether workers are enabled